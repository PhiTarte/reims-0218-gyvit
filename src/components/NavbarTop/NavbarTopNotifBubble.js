//TODO: create component notif bubble icon

import React from "react"
import PropTypes from "prop-types"
import { Badge } from "reactstrap"

const NavbarTopNotifBubble = ({ notifsCount }) => {
  return (
    <Badge
      style={{
        marginTop: "-50px",
        backgroundColor: "#ff007e",
        width: "35px",
        height: "20px"
      }}
      pill
    >
      {notifsCount}
    </Badge>
  )
}

NavbarTopNotifBubble.propTypes = {
  notifsCount: PropTypes.number.isRequired
}

export default NavbarTopNotifBubble
