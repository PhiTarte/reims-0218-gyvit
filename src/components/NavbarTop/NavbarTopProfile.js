//TODO: create component Profile

import React from "react"
import { Media } from "reactstrap"
import PropTypes from "prop-types"
import "bootstrap/dist/css/bootstrap.min.css"
import "./styles-navbar-top/navbarProfile.css"

const NavbarTopProfile = ({ firstname, lastname, business }) => (
  <div>
    <Media>
      <Media href="#">
        <Media
          object
          className="pic-profile rounded-circle"
          src="img/kevinMarlot.jpeg"
          alt={firstname}
        />
      </Media>
      <Media body>
        <Media className="id-profile" heading>
          <span>{firstname}</span> <span>{lastname}</span>
        </Media>

        <Media className="business-profile">{business}</Media>
      </Media>
    </Media>
  </div>
)

NavbarTopProfile.propTypes = {
  lastname: PropTypes.string.isRequired,
  firstname: PropTypes.string.isRequired,
  business: PropTypes.string.isRequired
}

export default NavbarTopProfile
