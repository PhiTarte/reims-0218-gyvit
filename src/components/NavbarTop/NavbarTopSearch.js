//TODO: create component search bar
// 2 props: search a afficher dans value de l'input
//          onSearchType a appelé sur l'évenement onchange de l'input
// faire un SearchContainer, qui stockera dans son state, un search

import React from "react"
import { FormGroup, Input } from "reactstrap"
import Select from "react-select"
import "react-select/dist/react-select.css"

const NavbarTopSearch = ({ options, search, onSearchType }) => (
  <div>
    <FormGroup>
      <span className="fa fa-search" />
      <Input
        style={{
          marginTop: "22px",
          marginLeft: "6px",
          height: "46px",
          backgroundColor: "#fbfcfd",
          borderRadius: "2px",
          border: "none"
        }}
        type="search"
        name="name"
        id="exampleSearch"
        placeholder="Search..."
        onChange={onSearchType}
        value={search}
        options={options}
      />
    </FormGroup>
  </div>
)

export default NavbarTopSearch
