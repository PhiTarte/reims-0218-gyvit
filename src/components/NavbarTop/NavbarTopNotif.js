//TODO: create component notif

import React from "react"
import { NavItem } from "reactstrap"
import NavbarTopNotifBubble from "./NavbarTopNotifBubble"

const NavbarTopNotif = ({ notifsCount }) => {
  //console.log(notifsCount)
  return (
    <NavItem className="col-1">
      <img
        style={{
          marginTop: "37px"
        }}
        src={process.env.PUBLIC_URL + "img/icon_notification.png"}
        alt="iconNotif"
      />
      <NavbarTopNotifBubble notifsCount={notifsCount} />
    </NavItem>
  )
}

export default NavbarTopNotif
