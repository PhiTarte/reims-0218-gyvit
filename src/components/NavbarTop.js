//TODO: create navbar top with search bar, notification icon and profile.

import React from "react"
import { Nav } from "reactstrap"

import NavbarTopNotif from "./NavbarTopNotif"
import NavbarTopSearch from "./NavbarTopSearch"
import NavbarTopProfile from "./NavbarTopProfile"

const NavbarTop = () => (
  <Nav className="row">
    <NavbarTopSearch />
    <NavbarTopNotif />
    <NavbarTopProfile />
  </Nav>
)

export default NavbarTop
