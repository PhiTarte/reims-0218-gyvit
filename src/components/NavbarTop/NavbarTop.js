//TODO: create navbar top with search bar, notification icon and profile.

import React from "react"
import { Nav } from "reactstrap"
import NavbarTopNotif from "./NavbarTopNotif"
import NavbarTopSearch from "./NavbarTopSearch"
import NavbarTopProfile from "./NavbarTopProfile"

const NavbarTop = ({ notif, profile }) => (
  <Nav
    style={{
      height: "90px",
      backgroundColor: "#ffffff",
      border: "solid 1px #e5e8f4"
    }}
    className="row"
  >
    <NavbarTopSearch />
    <NavbarTopNotif {...notif} />
    <NavbarTopProfile {...profile} />
  </Nav>
)

export default NavbarTop
