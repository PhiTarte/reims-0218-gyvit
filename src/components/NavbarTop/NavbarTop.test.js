//TODO: test:  navbar top with search bar, notification icon and profile.

import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import NavbarTop from "./NavbarTop"

const props = {
  notif: {
    notifsCount: 6
  },
  profile: {
    lastname: "Marlot",
    firstname: "Kevin",
    business: "Meduza"
  }
}

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<NavbarTop {...props} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders correctly", () => {
  const tree = renderer.create(<NavbarTop {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
