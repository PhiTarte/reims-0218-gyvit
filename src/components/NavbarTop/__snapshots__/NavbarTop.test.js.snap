// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "border": "solid 1px #E5E8F4",
      "height": "90px",
    }
  }
>
  <div
    className="col-7"
  >
    <div>
      <div
        className="form-group"
      >
        <span
          className="fa fa-search"
        />
        <input
          className="form-control"
          id="exampleSearch"
          name="search"
          placeholder="Search..."
          style={
            Object {
              "backgroundColor": "#fbfcfd",
              "border": "none",
              "borderRadius": "2px",
              "height": "46px",
              "marginLeft": "6px",
              "marginTop": "22px",
            }
          }
          type="search"
        />
      </div>
    </div>
  </div>
  <div
    className="col-2"
  >
    <div
      className="col-1"
    >
      <img
        alt="iconNotif"
        src="img/icon_notification.png"
        style={
          Object {
            "marginTop": "37px",
          }
        }
      />
      <span
        className="badge badge-secondary badge-pill"
        style={
          Object {
            "backgroundColor": "#ff007e",
            "height": "20px",
            "width": "20px",
          }
        }
      >
        6
      </span>
    </div>
  </div>
  <div
    className="col-3"
  >
    <div>
      <div
        className="media"
      >
        <div
          className="media"
          href="#"
        >
          <img
            alt="Romain"
            className="rounded-circle media-object"
            src="http://via.placeholder.com/40x40"
            style={
              Object {
                "marginTop": "25px",
              }
            }
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="media-heading"
            style={
              Object {
                "color": "#231b56",
                "fontSize": "16px",
                "marginLeft": "21px",
                "marginTop": "28px",
              }
            }
          >
            <span>
              Romain
            </span>
             
            <span>
              Duflot
            </span>
          </h4>
          <div
            className="media"
            style={
              Object {
                "color": "#372c78",
                "fontSize": "14px",
                "marginLeft": "21px",
                "marginTop": "5px",
              }
            }
          >
            Notoriety
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
